addons/mail/static/src/models/chat_window_manager/chat_window_manager.js

```diff
@@ -124,6 +124,9 @@ function factory(dependencies) {
                     manager: [['link', this]],
                     thread: [['link', thread]],
                 });
+                if (thread.model !== 'mail.channel') {
+                    await thread.fetchData();
+                }
```

Comment:
I would make another route just to check right access.
`/mail/thread/data` can be quite heavy, so there's a lot of wasted processing to call this route just to check and update access rights.